### ISSUE TITLE
`Brave.Rewards.WalletState` doesn't record values 3 & 4

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#16483]
## Description 
Found while testing #7634, `Brave.Rewards.WalletState` doesn't record values 3 & 4

## Steps to reproduce 

   1. Clean install
   2. Switch to staging flag, do not claim grant
   3. Connect to Uphold account which has funds
   4. Wait for 10 sec, restart browser if required
   5. Brave.Rewards.WalletState stays at value 1
   6. Claim UGP grant 
   7. Wait and check value still remains at 1

## Actual result 
`Brave.Rewards.WalletState` doesn't record values 3 & 4

## Expected result
Should update accordingly 

## Issue reproduces how often 
Easy

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? NA
- Can you reproduce this issue with the current Play Store Beta version? Yes
- Can you reproduce this issue with the current Play Store Nightly version? Yes

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): All
   - Android version: All
   
## Brave version
1.26.63

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? NA
- Does the issue resolve itself when disabling Brave Rewards? NA
- Is the issue reproducible on the latest version of Chrome? NA

### Additional information

cc: @GeetaSarvadnya @emerick

**Comment by GeetaSarvadnya** [comment]
Reproduced on Desktop also:

```


<!--StartFragment-->
Brave | 1.26.60 Chromium: 91.0.4472.101&nbsp;(Official Build)&nbsp;beta&nbsp;(64-bit)
-- | --
Revision | af52a90bf87030dd1523486a1cd3ae25c5d76c9b-refs/branch-heads/4472@{#1462}
OS | Windows&nbsp;10 OS Version 2004 (Build 19041.1052)

<!--EndFragment-->


```

**Comment by GeetaSarvadnya** [comment]
Good find @srirambv !. 

If we fund the wallet via restore wallet the P3A walletstate values are shown correctly. But when the wallet is funded via uphold the values are shown incorrectly. I am not sure how important this P3A value from a product point of view. Can we fix this issue as part of 1.26.x or defer to the next release?
cc: @bsclifton @emerick @rebron

**Comment by Miyayes** [comment]
Question has been removed, so no longer relevant.
